src/dnx/dnx.cpp

```diff
@@ -31,9 +31,9 @@ bool StringsEqual(LPCWSTR const pszStrA, LPCWSTR const pszStrB)
 
 bool StringEndsWith(LPCWSTR const pszStr, LPCWSTR const pszSuffix)
 {
-    int nStrLen = wcsnlen_s(pszStr, MAX_PATH);
-    int nSuffixLen = wcsnlen_s(pszSuffix, MAX_PATH);
-    int nOffset = nStrLen - nSuffixLen;
+    size_t nStrLen = wcsnlen_s(pszStr, MAX_PATH);
+    size_t nSuffixLen = wcsnlen_s(pszSuffix, MAX_PATH);
+    size_t nOffset = nStrLen - nSuffixLen;
```

Comment:
Making this size_t will cause the below if statement to always be false, and might cause other issues too.